In [2]:
import os
import json
import gzip
import pandas as pd

In [3]:
!pwd

/Users/auraulloa/Desktop/DataSciencePipeline/project


# CD and Vinyls

In [4]:
path = '/Users/auraulloa/Desktop/DataSciencePipeline/'
physical_path = path + 'meta_CDs_and_Vinyl.json.gz'

In [5]:
### load the meta data

phys_data = []
with gzip.open(physical_path) as f:
    for l in f:
        phys_data.append(json.loads(l.strip()))
    
# total length of list, this number equals total number of products
print(len(phys_data))

516914


There are 516,914 physical products.

In [6]:
# convert list into pandas dataframe

phys_df = pd.DataFrame.from_dict(phys_data)

print(len(phys_df))

516914


In [7]:
phystemp = phys_df.fillna('')
physicals = phystemp[~phystemp.title.str.contains('getTime')] # filter those unformatted rows
print(len(physicals))

516914


In [8]:
physicals.head()

category tech1  \
0                      [CDs & Vinyl, Christian]         
1  [CDs & Vinyl, Christian, Pop & Contemporary]         
2          [CDs & Vinyl, Christian, Children's]         
3          [CDs & Vinyl, Christian, Children's]         
4          [CDs & Vinyl, Christian, Children's]         

                                         description fit  \
0                                                 []       
1                                                 []       
2                                                 []       
3  [1) To God Be The Glory2) This Is My Fathers W...       
4  [Joy To The World 2 The First Noel 3 Hark! The...       

                                  title  \
0                Songs for the Shepherd   
1       Lift Him Up With Ron Kenoly VHS   
2                            I Love You   
3  Hymns: 16 Classic Hymns for Children   
4                      Christmas Carols   

                                            also_buy tech2            brand  \
0  [B0016CP2GS, B0000075PB, B0000275QQ, B00000I7J...            Keith Green   
1               [6303908845, 6303646611, B000008GO1]                          
2                                                 []        Integrity Music   
3  [B00000DPLJ, B000008UPG, B00BQH9RKA, B00KYVHGR...                Sue Gay   
4                           [B0010WAC50, B00000DPLJ]         Cedarmont Kids   

  feature                        rank  \
0      []    123,896 in CDs & Vinyl (   
1      []    284,935 in Movies & TV (   
2      []  1,138,833 in CDs & Vinyl (   
3      []    271,064 in CDs & Vinyl (   
4      []    409,293 in CDs & Vinyl (   

                                           also_view  \
0  [B0016CP2GS, B00000I7JO, B00000I9AH, B00P826ZR...   
1                                       [B00000DUDE]   
2                                                 []   
3  [B00000DPLL, 0843113006, B00BQH9RKA, B000008UP...   
4  [B00000DPKG, B0010WAC50, B00000DPKH, B00FSZPM8...   

                                            main_cat similar_item date  \
0  <img src="https://images-na.ssl-images-amazon....                     
1                                        Movies & TV                     
2  <img src="https://images-na.ssl-images-amazon....                     
3  <img src="https://images-na.ssl-images-amazon....                     
4  <img src="https://images-na.ssl-images-amazon....                     

                                               price        asin imageURL  \
0                                                     0001393774       []   
1                                                     0001501348       []   
2                                                     0001515209       []   
3  .a-section.a-spacing-mini{margin-bottom:6px!im...  0005072298       []   
4                                                     0005121515       []   

  imageURLHighRes details  
0              []          
1              []          
2              []          
3              []          
4              []

# Digital Music

In [9]:
### load the meta data
dig_path = path + 'meta_Digital_Music.json.gz'
dig_data = []
with gzip.open(dig_path) as f:
    for l in f:
        dig_data.append(json.loads(l.strip()))
    
# total length of list, this number equals total number of products
print(len(dig_data))

74347


There are 74,347 digital music products in dataset. Much less the 516,914 physical ones in the set.

In [10]:
# convert list into pandas dataframe

dig_df = pd.DataFrame.from_dict(dig_data)

print(len(dig_df))

74347


In [11]:
digtemp = dig_df.fillna('')
digitals = digtemp[~digtemp.title.str.contains('getTime')] # filter those unformatted rows
print(len(digitals))

74347


In [12]:
digitals.head()

category tech1                                        description fit  \
0       []                                                       []       
1       []                                                       []       
2       []                                                       []       
3       []                                                       []       
4       []        [1. Losing Game 2. I Can't Wait 3. Didn't He S...       

                               title  \
0       Master Collection Volume One   
1  Hymns Collection: Hymns 1 &amp; 2   
2        Early Works - Don Francisco   
3      So You Wanna Go Back to Egypt   
4          Early Works - Dallas Holm   

                                            also_buy tech2  \
0  [B000002UEN, B000008LD5, B01J804JKE, 747403435...         
1                           [5558154950, B00014K5V4]         
2  [B00004RC05, B003H8F4NA, B003ZFVHPO, B003JMP1Z...         
3  [B0000275QQ, 0001393774, 0001388312, B0016CP2G...         
4  [B0002N4JP2, 0760131694, B00002EQ79, B00150K8J...         

                    brand feature                      rank  \
0     John Michael Talbot      []   58,291 in CDs & Vinyl (   
1  Second Chapter of Acts      []   93,164 in CDs & Vinyl (   
2           Don Francisco      []  875,825 in CDs & Vinyl (   
3             Keith Green      []  203,263 in CDs & Vinyl (   
4             Dallas Holm      []  399,269 in CDs & Vinyl (   

                                           also_view  \
0  [B000002UEN, B000008LD5, 7474034352, B000008LD...   
1  [B000008KJ3, B000008KJ0, 5558154950, B000UN8KZ...   
2  [B003H8F4NA, B003ZFVHPO, B003JMP1ZK, B00004RC0...   
3  [B00000I7JO, B0016CP2GS, 0001393774, B0000275Q...   
4  [B0002N4JP2, 0760131694, B00150K8JC, B003MTXNV...   

                                            main_cat similar_item date  \
0  <img src="https://images-na.ssl-images-amazon....                     
1  <img src="https://images-na.ssl-images-amazon....                     
2  <img src="https://images-na.ssl-images-amazon....                     
3  <img src="https://images-na.ssl-images-amazon....                     
4  <img src="https://images-na.ssl-images-amazon....                     

    price        asin imageURL imageURLHighRes details  
0  $18.99  0001377647       []              []          
1          0001529145       []              []          
2          0001527134       []              []          
3  $13.01  0001388703       []              []          
4          0001526146       []              []

In [13]:
issa_match = physicals[physicals['title'].isin(digitals['title']) == True]
len(issa_match)

45532

In [14]:
issa_match2 = digitals[digitals['title'].isin(physicals['title']) == True]
len(issa_match2)

12983

Physical
df5: 516,914

Digital
df9: 74,347

Merge 45,532 (when checking if physical copies are in digital copies)
Only 12,983 when doing it the other way :(

can't connect through asin much like the other dataset...

In [16]:
len(issa_match)/len(digitals)

0.6124255181782722

+60% of digital products get matched up to physical ones.

In [23]:
#dfa = digitals.drop_duplicates(subset=['title'])
#dfb = physicals.drop_duplicates(subset=['title'])

try_1 = pd.merge(physicals, how='inner', on='title')

In [24]:
len(try_1)

33202700

5184 dfa, dfb, drop duplicates
33202700, digitals, physicals